TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/fragment/PostFragment.java

```diff
@@ -0,0 +1,375 @@
+    }
+
+    private void setPostButton(View view) {
+        // Post button
+        view.findViewById(R.id.button_post).setOnClickListener(v -> {
+
+            // Check text length
+            int length = mPostEdit.getText().length();
+            if (length == 0 && mImageUris.isEmpty()) {
+                return;
+            }
+            // Prepare data
+            long replyId = mReplyStatus != null ? mReplyStatus.getId() : -1;
+            String quoteURL = mQuoteStatus != null
+                    ? " https://twitter.com/" + mQuoteStatus.getUser().getScreenName() + "/status/" + mQuoteStatus.getId()
+                    : "";
+
+            // Create tweet, then post
+            StatusUpdate update = new StatusUpdate(
+                    mPostEdit.getText().toString() + quoteURL
+            );
+            update.setInReplyToStatusId(replyId);
+            StatusUseCase.post(update, mImageUris);
+
+            // Save hashtag
+            EntityArray<String> hashtags = TweetMateApp.getMyAccount().getHashtags();
+            String regex = "[#＃](w*[一-龠_ぁ-ん_ァ-ヴーａ-ｚＡ-Ｚa-zA-Z0-9]+|[a-zA-Z0-9_]+|[a-zA-Z0-9_]w*)";
+            Matcher matcher = Pattern.compile(regex).matcher(mPostEdit.getText());
+            while (matcher.find()) {
+                if (hashtags.size() >= 8) {
+                    hashtags.remove(0);
+                }
+                hashtags.add(matcher.group());
+            }
+        });
+    }
+
+    private void setUserButton(View view) {
+        // Get user
+        TwitterUser user = TweetMateApp.getMyUser();
+        ImageView userButton = view.findViewById(R.id.button_user);
+
+        // Set user image
+        ImageOption option = ImageOption.toEnum(PrefAppearance.getUserIconStyle());
+        GlideUtils.load(PrefSystem.getProfileThumbByQuality(user), userButton, option);
+
+        // User button
+        userButton.findViewById(R.id.button_user).setOnClickListener(v -> {
+            // Coming soon...
+        });
+    }
+
+    private void setGalleryButton(View view) {
+        // Gallery button
+        view.findViewById(R.id.button_gallery).setOnClickListener(v -> {
+            if (!isPicAvailable()) {
+                return;
+            }
+            // Open gallery
+            Intent intent = new Intent(Intent.ACTION_PICK);
+            intent.setType("image/*");
+            getActivity().startActivityForResult(intent, REQUEST_GALLERY);
+        });
+    }
+
+    private void setDraftButton(View view) {
+        // Draft button
+        view.findViewById(R.id.button_draft).setOnClickListener(v -> {
+
+            // Check drafts
+            EntityArray<String> drafts = TweetMateApp.getMyAccount().getDrafts();
+            if (drafts.isEmpty()) {
+                NoticeDialog dialog = new NoticeDialog().newInstance(STR_FAIL_NO_DRAFT);
+                FragmentManager fm = getActivity().getSupportFragmentManager();
+                if (fm.findFragmentByTag("NoticeDialog") == null) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "NoticeDialog" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1192)